indev/libinput.c

```diff
@@ -259,37 +353,24 @@ void libinput_read(lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
  */
 void libinput_read_state(libinput_drv_state_t * state, lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
 {
-  struct libinput_event *event;
-  int rc = 0;
+  LV_UNUSED(indev_drv);
 
-  rc = poll(state->fds, nfds, timeout);
-  switch (rc){
-    case -1:
-      perror(NULL);
-    case 0:
-      goto report_most_recent_state;
-    default:
-      break;
+  pthread_mutex_lock(&state->event_lock);
+  if (!event_pending(state)) {
```

Comment:
I think you can remove this `if` because `get_event` does the same test and returns `NULL` if it fails which is then skipped by the `if (evt)` below and ends up in the same unlock and return behavior.